modin/pandas/concat.py

```diff
@@ -132,3 +144,34 @@ def concat(
         else:
             result_df.columns = new_idx
     return result_df
+
+
+def _determine_name(objs: list, axis):
+    """
+    Determine names of index after concatenation along passed axis
+
+    Parameters
+    ----------
+    objs : list of DataFrames or QueryCompilers
```

Comment:
For simplicity we should probably just either allow `objs` to be `DataFrame` or `QueryCompiler` object, but not both.

Reply:
In a current implementation, `concat` will call `_determine_name` only with `QueryCompiler`, but because `QueryCompiler` is partially repeating interface of `DataFrame`, that functions can work well and with DFs too. Should we change documentation in that case?

Reply:
Yes, if it only accepts `QueryCompiler` I believe we should document that. It is an internal functionality so we can control it.

Reply:
done